src/hooks/useAgentZeroStatusIndicator.ts

```diff
@@ -0,0 +1,252 @@
+import {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import ONYXKEYS from '@src/ONYXKEYS';
+import useOnyx from './useOnyx';
+
+type AgentZeroStatusState = {
+    displayLabel?: string;
+    kickoffWaitingIndicator: () => void;
+};
+
+const WAITING_LABEL = 'Concierge is waiting for you to finish...';
+const INITIAL_FILLER_DELAY_MS = 4000;
+const MAX_FILLER_DELAY_MS = 8000;
+const BACKOFF_MULTIPLIER = 1.8;
+const JITTER_RATIO = 0.25; // +/-25% jitter to keep timers from feeling scripted
+
+const GENERIC_FILLER_MESSAGES = [
+    'Concierge is gathering context...',
+    'Concierge is analyzing semantic content...',
+    'Concierge is considering responses...',
+    'Concierge is verifying details...',
+    'Concierge is organizing your information...',
+    'Concierge is identifying key insights...',
+    'Concierge is synthesizing relevant data...',
+    'Concierge is checking for potential issues...',
+    'Concierge is refining its understanding...',
+    'Concierge is preparing a response strategy...',
+    'Concierge is reviewing edge cases...',
+    'Concierge is aligning on the best answer...',
+    'Concierge is ensuring clarity and accuracy...',
+];
```

Comment:
Do these strings need translations, or are we not worrying about that for concierge yet?